scripts/solve.js

```diff
@@ -243,3 +248,31 @@ function highlightCode(interpreter) {
     lastMarking.clear();
   lastMarking = codeMirror.getDoc().markText(start, end, {className: 'highlighted'});
 }
+
+/**
+ * Saves current code with current location as a key.
+ **/
+function sessionStoreCode() {
+  let key   = window.location.href;
+  let value = codeMirror.getValue();
+  window.sessionStorage.setItem(key, value);
```

Comment:
should we consider localStorage rather than sessionStorage? a case i'm thinking of is that if someone accidentally closes the tab they're working on it would be nice for that information to be stored.

Reply:
Good point!
Updated it localStorage :)

Reply:
Good point!
Updated it to localStorage :)